### ISSUE TITLE
PhotonGI + lightgroups + clamping leads to fireflies

### ISSUE THREAD

**Issue by Theverat** [LuxCoreRender/LuxCore#328]
OS: Windows 10
LuxCore version: 91c85d51f2b1fafcb87bbb8d5cc9d670bd93bd86
Scene: [photongi_clamping_lightgroups_fireflies.zip](https://github.com/LuxCoreRender/LuxCore/files/4432145/photongi_clamping_lightgroups_fireflies.zip)

In this specific scene, the combination of PhotonGI + lightgroups + clamping leads to fireflies on a glossy translucent material.
Happens on CPU and on GPU.
Not sure what exactly is going on here, and if it even is a bug.

This is the "sun" light group with PhotonGI + clamping:
![sun_pgi_clamping](https://user-images.githubusercontent.com/1584761/78458075-13b1df80-76af-11ea-816a-f873b97d1b61.png)

"sun" light group with just PhotonGI:
![sun_pgi](https://user-images.githubusercontent.com/1584761/78458079-1a405700-76af-11ea-966c-e7bec554ac8c.png)

"sun" light group with just clamping:
![sun_clamping](https://user-images.githubusercontent.com/1584761/78458083-1e6c7480-76af-11ea-8008-c6f25174a375.png)

"sun" light group without clamping or PhotonGI:
![sun](https://user-images.githubusercontent.com/1584761/78458091-24faec00-76af-11ea-88ae-0517d8afd827.png)

The above are renders from the attached stripped-down version of the Danish Mood scene.
Below are the two renders where I originally spotted the problem:

PhotonGI + clamping:
![with_clamping](https://user-images.githubusercontent.com/1584761/78458164-7f944800-76af-11ea-90e5-fc8d4199b3a4.png)

PhotonGI, no clamping:
![without_clamping](https://user-images.githubusercontent.com/1584761/78458178-8ae77380-76af-11ea-8235-ec35f75eeac5.png)

**Comment by Dade916** [comment]
The source of the problem is here:

`scene.lights.140526408646152.radius = 0.0600000024`

The radius of the sun i reduced by about 20 times. It is already very very small by default and very hard to find for a path tracer.

Reducing the radius by 20 times starts to produce ballistic numbers. It just not make any sense as setting.

For what I observed, it is not related to PGI or clamping at all (may be clamping goes ballistics as side effect of the numbers produced when hitting the sun disk).

Short version: it is a numerical precision problem for very small radius sizes.

**Comment by Theverat** [comment]
> The source of the problem is here:
`scene.lights.140526408646152.radius = 0.0600000024`

I think you have mixed up two lights there. The sun size is set with the `relsize` property, not `radius` - the latter belongs to the sphere light in the scene.

The properties look like this here. I inserted a blank line between the two lights, due to the bad naming it is very hard to distinguish them otherwise:
```
# sphere light
scene.lights.3158114469736.gain = 200 200 200
scene.lights.3158114469736.importance = 1
scene.lights.3158114469736.id = 2
scene.lights.3158114469736.type = "sphere"
scene.lights.3158114469736.color = 1 1 1
scene.lights.3158114469736.efficency = 0
scene.lights.3158114469736.power = 0
scene.lights.3158114469736.normalizebycolor = 0
scene.lights.3158114469736.position = 0 0 0
scene.lights.3158114469736.transformation = 1 0 0 0 0 1 0 0 0 0 1 0 -1.4775300025939941 1.5820077657699585 2.0828022956848145 1
scene.lights.3158114469736.radius = 0.060000002384185791
scene.lights.3158114469736.visibility.indirect.diffuse.enable = 1
scene.lights.3158114469736.visibility.indirect.glossy.enable = 1
scene.lights.3158114469736.visibility.indirect.specular.enable = 1

# sun
scene.lights.3158114475688.gain = 1.9999999494757503e-05 1.9999999494757503e-05 1.9999999494757503e-05
scene.lights.3158114475688.importance = 1
scene.lights.3158114475688.id = 3
scene.lights.3158114475688.type = "sun"
scene.lights.3158114475688.dir = 0.65174049139022827 -0.64096939563751221 0.40545362234115601
scene.lights.3158114475688.turbidity = 2.2000000476837158
scene.lights.3158114475688.relsize = 1
scene.lights.3158114475688.visibility.indirect.diffuse.enable = 1
scene.lights.3158114475688.visibility.indirect.glossy.enable = 1
scene.lights.3158114475688.visibility.indirect.specular.enable = 1
```
